sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/PerformanceCounters/PerformanceCounterCollectorFactory.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using Azure.Monitor.OpenTelemetry.Exporter.Internals.Platform;
+using Azure.Monitor.OpenTelemetry.LiveMetrics.Internals.Diagnostics;
+
+namespace Azure.Monitor.OpenTelemetry.LiveMetrics.Internals.PerformanceCounters
+{
+    internal static class PerformanceCounterCollectorFactory
+    {
+        private const string WEBSITE_ISOLATION_HYPERV = "hyperv";
+
+        public static bool TryGetInstance(IPlatform platform, out IPerformanceCounterCollector? performanceCounterCollector)
+        {
+            try
+            {
+                var isWindows = platform.IsOSPlatform(OSPlatform.Windows);
+                var isAzureAppService = IsAzureAppService(platform);
+
+                if (isAzureAppService && isWindows)
+                {
+                    performanceCounterCollector = new AzureWebAppWindowsPerformanceCounterCollector();
+                }
+                else if (isWindows)
+                {
+                    performanceCounterCollector = new WindowsPerformanceCounterCollector();
+                }
+                else
+                {
+                    performanceCounterCollector = new NonWindowsPerformanceCounterCollector();
+                }
+
+                LiveMetricsExporterEventSource.Log.PerformanceCounterCollectorFactoryDecision(isWindows, isAzureAppService, performanceCounterCollector.GetType().Name);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                // Since these are platform specific classes, exceptions could be thrown during constructor call.
+                LiveMetricsExporterEventSource.Log.PerformanceCounterCollectorFactoryFailed(ex);
+                performanceCounterCollector = null;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Evaluates environment variables specific to Azure Web App.
+        /// </summary>
+        /// <remarks>
+        /// Presence of "WEBSITE_SITE_NAME" indicate web apps.
+        /// "WEBSITE_ISOLATION"=="hyperv" indicate premium containers.
+        /// In the case of premium containers, perf counters can be read using regular mechanism and hence this method returns false.
+        /// </remarks>
+        private static bool IsAzureAppService(IPlatform platform)
```

Comment:
Do we have this logic somewhere else in exporter or distro. I think it should be a part of sdm metrics.

Reply:
We could just rely on WEBSITE_SITE_NAME

Reply:
It would be simpler if we can just rely on WEBSITE_SITE_NAME.
Take note of the comment on line 52-53, I found that message in AI SDK.
Have not independently verified yet.

Reply:
Removed.